php/class-photonfill.php

```diff
@@ -688,14 +681,47 @@ public function get_attachment_picture( $attachment_id, $size = 'full', $attr )
 						}
 
 						// Set our default img element
-						$html .= '<img srcset="' . esc_url( $default_srcset ) . '"' . $alt . ' ' . $style . '>';
+						$attr['srcset'] = $default_srcset;
+						$html .= $this->build_attachment_image( $attachment_id, $attr );
 						$html .= '</picture>';
 					}
 				}
 				return $html;
 			}
 			return;
 		}
+		
+		/**
+		 * Get the HTML for the IMG Tag
+		 *
+		 * @param $attachment_id int Attachment ID
+		 * @param $attr array Image Attributes
+		 *
+		 * @return string
+		 */
+		private function build_attachment_image( $attachment_id, $attr ) {
+			$attachment = get_post( $attachment_id );
+			
+			if ( empty( $attr['alt'] ) ) {
+				$attr['alt'] = trim( strip_tags( get_post_meta($attachment_id, '_wp_attachment_image_alt', true ) ) );
+			}
+			if ( empty($attr['alt'] ) ) {
+				$attr['alt'] = trim( strip_tags( $attachment->post_excerpt ) ); // If not, Use the Caption
+			}
+			if ( empty($attr['alt'] ) ) {
+				$attr['alt'] = trim( strip_tags( $attachment->post_title ) ); // Finally, use the title
+			}
+			$html = '<img ';
+			foreach ( $attr as $key => $value ) {
+				if ( is_bool( $value ) && $value ) {
+					$html .= esc_attr( $value ) . ' ';
```

Comment:
Don't we mean `$html .= esc_attr( $key ) . ' ';`  here? I'm not sure I understand this otherwise.